addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/RefreshStrategy.java

```diff
@@ -23,17 +26,45 @@
  */
 @NonNullByDefault
 public class RefreshStrategy {
-    private static final int DEFAULT_DELAY = 30;
+
+    private Logger logger = LoggerFactory.getLogger(RefreshStrategy.class);
+
+    private static final int DEFAULT_DELAY = 30; // in seconds
+    private static final int SEARCH_REFRESH_INTERVAL = 120; // in seconds
     private int dataValidityPeriod;
     private long dataTimeStamp;
+    private boolean searchRefreshInterval;
+    @Nullable
+    private Integer dataTimestamp0;
 
     // By default we create dataTimeStamp to be outdated
+    // A null or negative value for dataValidityPeriod will trigger an automatic search of the validity period
     public RefreshStrategy(int dataValidityPeriod) {
-        this.dataValidityPeriod = dataValidityPeriod;
+        if (dataValidityPeriod <= 0) {
+            this.dataValidityPeriod = 0;
+            this.searchRefreshInterval = true;
+            logger.debug("Data validity period search...");
+        } else {
+            this.dataValidityPeriod = dataValidityPeriod;
+            this.searchRefreshInterval = false;
+            logger.debug("Data validity period set to {} ms", this.dataValidityPeriod);
+        }
         expireData();
     }
 
     public void setDataTimeStamp(Integer dataTimestamp) {
+        if (searchRefreshInterval) {
+            if (dataTimestamp0 == null) {
+                dataTimestamp0 = dataTimestamp;
+                logger.debug("First data timestamp is {}", dataTimestamp0);
+            } else if (dataTimestamp.intValue() > dataTimestamp0.intValue()) {
+                dataValidityPeriod = (dataTimestamp.intValue() - dataTimestamp0.intValue()) * 1000;
```

Comment:
I see a lot of mapping between MS en S, maybe we could just (I know breaking) consider changing it to seconds.